### ISSUE TITLE
Pre-rollout tasks fail when an environment is idled in kubernetes

### ISSUE THREAD

**Issue by shreddedbacon** [amazeeio/lagoon#2594]
# Describe the bug

If an environment is idled in kubernetes by the new lagoon-idler, there is no way to unidle the environment without hitting the ingress first, or ssh-ing into the environment.

This can be problematic for deployments that have `pre-rollout` tasks that may need to talk to an idled resource.

## To Reproduce

1. Wait for an environment to idle.
2. Trigger a build that contains a pre-rollout task that contacts a potentially idled deployment.
3. Observe the build failure as it tries to contact something that is idled.

## Expected behavior

The environment should be unidled if there are any pre-rollout tasks to perform.

## Additional context

This problem does not exist in openshift as idling is done differently, and only recently idling was rolled out to all kubernetes clusters so this has only been discovered recently.

**Comment by shreddedbacon** [comment]
PR https://github.com/amazeeio/lagoon/pull/2583 was merged which unidles environments when an SSH connection is established, so running something like this will bring the environment back up.
```lagoon ssh -p $project -e $env -C "echo unidle"```

I propose adding the unidling logic from that PR into the kubernetes (not openshift) `build-deploy-docker-compose.sh` into just before the pre-rollout tasks are run https://github.com/amazeeio/lagoon/blob/master/images/kubectl-build-deploy-dind/build-deploy-docker-compose.sh#L475

**Comment by tobybellwood** [comment]
I agree, the kubectl should unidle before a deployment!

**Comment by smlx** [comment]
Is this issue still valid?

**Comment by shreddedbacon** [comment]
Yes, we only unidle the specific service that is called for pre-rollout tasks, not the entire environment: https://github.com/uselagoon/lagoon/blob/main/images/kubectl-build-deploy-dind/scripts/exec-pre-tasks-run.sh
